### PR TITLE
uavobjectbrowser: handle highlight notifications nicely

### LINKED ISSUE #1775: Highlights in UAV browser broken
My fix in the static analysis PR, and subsequently sucked.  Need to look at, but can't now.

### PR DESCRIPTION
**Pull request by mlyle**
Fixes #1775 .  Also makes things a lot more sane and avoids a whole lot of random pinging events around.

On my Linux laptop-- GCS connected to sim, on config pane, with "Data objects" expanded-- with old code uses 28.5% CPU.  With this code uses 25.5% CPU.

### CODE REVIEW

**Comment by tracernz** [comment]
Quick glance, looks good. 😁 Will read it properly..

File: ground/gcs/src/plugins/uavobjectbrowser/treeitem.cpp [hunk 1]
```diff
@@ -83,24 +75,26 @@ bool HighLightManager::remove(TreeItem *itemToRemove)
...
-        TreeItem *item = iter.next();
-        if (item->getHiglightExpires() < *m_currentTime) {
-            // If expired, call removeHighlight
-            item->removeHighlight();
-
-            // Remove from list since it is restored.
-            iter.remove();
-        }
+    for (QSet<TreeItem *>::const_iterator i = m_itemsToUnhighlight.begin();
+            i != m_itemsToUnhighlight.end(); ++i) {
```
Comment (tracernz) [inline ground/gcs/src/plugins/uavobjectbrowser/treeitem.cpp:79]: No ranged-for love? :stuck_out_tongue:

File: ground/gcs/src/plugins/uavobjectbrowser/treeitem.cpp [hunk 2]
```diff
@@ -57,22 +50,21 @@ HighLightManager::HighLightManager(long checkingInterval, QTime *currentTime)
...
- * Called to remove item from list.
- * Returns true if item was removed, otherwise false.
- */
-bool HighLightManager::remove(TreeItem *itemToRemove)
-{
-    // Remove item and return result
-    return m_itemsList.removeOne(itemToRemove);
+    if (m_itemsWaiting.contains(itemToAdd)) {
+        return false;
+    }
```
Comment (tracernz) [inline ground/gcs/src/plugins/uavobjectbrowser/treeitem.cpp:59]: I like this style even less in GCS :stuck_out_tongue:

**Review by tracernz — APPROVED** [review]
A shame to lose the setting. Code looks good though. :+1:

File: ground/gcs/src/plugins/uavobjectbrowser/treeitem.cpp (hunk 1 above)
Comment (mlyle) [reply ground/gcs/src/plugins/uavobjectbrowser/treeitem.cpp:79]: Some syntax is just too fancy for me :<

**Pull request merged**